src/Workspaces/Core/MSBuild.BuildHost/Build/ProjectBuildManager.cs

```diff
@@ -158,18 +161,21 @@ public void StartBatchBuild(IDictionary<string, string>? globalProperties = null
 
             globalProperties ??= ImmutableDictionary<string, string>.Empty;
             var allProperties = s_defaultGlobalProperties.RemoveRange(globalProperties.Keys).AddRange(globalProperties);
-            _batchBuildProjectCollection = new MSB.Evaluation.ProjectCollection(allProperties);
+
             _batchBuildLogger = new MSBuildDiagnosticLogger()
             {
                 Verbosity = MSB.Framework.LoggerVerbosity.Normal
             };
 
+            var loggers = _msbuildLogger is not null
+                ? ImmutableArray.Create(_msbuildLogger)
+                : ImmutableArray<MSB.Framework.ILogger>.Empty;
+
+            _batchBuildProjectCollection = new MSB.Evaluation.ProjectCollection(allProperties, loggers, MSB.Evaluation.ToolsetDefinitionLocations.Default);
```

Comment:
Should we also be adding the _batchBuildLogger to the collection here as well? If not, a comment explaining why might still be good. The fact there's also not one being created in the non-batch flow earlier to me is unclear.

Reply:
> Should we also be adding the _batchBuildLogger to the collection here as well?

No - otherwise the _batchBuildLogger logger will get initialized twice (and throw).  We also already collect the diagnostic log separately for project evaluation, so it isn't necessary to pass it in.  Can add a comment.

> The fact there's also not one being created in the non-batch flow earlier to me is unclear

As I understand it, the non-batch flow evaluation is line 126, where I also add the logger.  This adds the logger to the project collection in the batch flow.

Reply:
Note - I cannot find anywhere that does a build outside of batch build in this file - so I don't think there is any other non-batch build to add a logger to.  The build steps assert that a batch build was started.

Looks like its only the project evaluation that can happen outside of batch build.